### PR TITLE
Remove review type selector in upload and option to request review.

### LINKED ISSUE #3085: Remove review process selection step and request full review link
This is part of #3083.

Step 6, the review process selection step, should be removed from the submission flow, for both listed and unlisted add-ons. They should all go to the Full Review track by default.

In the Developer Hub, the Request Full Review link that appears for add-ons with Preliminary Review should also be removed.

### PR DESCRIPTION
**Pull request by eviljeff**
Fixes #3085

### CODE REVIEW

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -1213,10 +1213,22 @@ def thumbnail_url(self):
         except IndexError:
             return settings.STATIC_URL + '/img/icons/no-preview.png'
 
-    def can_request_review(self):
+    def can_request_review(self, disallow_preliminary_review=False):
         """Return the statuses an add-on can request."""
         if not File.objects.filter(version__addon=self):
             return ()
+        if (disallow_preliminary_review):
```
Comment (EnTeQuAk) [inline src/olympia/addons/models.py]: I don't think we need the parens here

File: src/olympia/addons/models.py [hunk 2]
```diff
@@ -1213,10 +1213,22 @@ def thumbnail_url(self):
...
+            if (self.is_disabled or
+                    self.status in (amo.STATUS_PUBLIC,
+                                    amo.STATUS_NOMINATED,
+                                    amo.STATUS_DELETED) or
+                    not self.latest_version or
+                    not self.latest_version.files.exclude(
+                        status=amo.STATUS_DISABLED)):
```
Comment (diox) [inline src/olympia/addons/models.py]: No need to load the objects here, so add a `.exists()`

File: src/olympia/addons/tests/test_models.py [hunk 3]
```diff
@@ -1280,18 +1280,18 @@ def test_can_request_review_rejected(self):
...
         assert addon.can_request_review() == ()
 
-    def check(self, status, exp, kw={}):
+    def check(self, status, exp, no_prelim=False, kw={}):
         addon = Addon.objects.get(pk=3615)
         changes = {'status': status, 'disabled_by_user': False}
         changes.update(**kw)
         addon.update(**changes)
-        assert addon.can_request_review() == exp
+        assert addon.can_request_review(no_prelim) == exp
```
Comment (diox) [inline src/olympia/addons/tests/test_models.py]: Use the keyword argument for clarity ? 

`assert addon.can_request_review(disallow_preliminary_review=no_prelim) == exp` 

Maybe even rename `no_prelim` to `disallow_preliminary_review` as well.

Comment (diox) [inline src/olympia/addons/tests/test_models.py]: Actually, I know it was already there but I really hate that `check` method. By looking at the calls below you have no idea what it's doing.

I suggest changing its signature to use keyword arguments and use those keyword arguments in every call below. Something like this:

```
def check(self, status=None,  expected_statuses=None, disallow_preliminary_review=False,
          extra_update_kwargs={}):
    if status is None or  expected_statuses is None:
        raise ImproperlyConfiguredError
    addon = Addon.objects.get(pk=3615)
    changes = {'status': status, 'disabled_by_user': False}
    changes.update(**extra_update_kwargs)
    addon.update(**changes)
    assert addon.can_request_review() == expected_statuses
    assert addon.can_request_review(disallow_preliminary_review) ==  expected_statuses
```

File: src/olympia/devhub/templates/devhub/addons/submit/sidebar.html [hunk 4]
```diff
@@ -4,13 +4,23 @@
...
    to listed or unlisted addons, a class of 'listed' means it's only relevant
    to listed addons, not unlisted ones. #}
-{% set NAV = ((_('Getting Started'), 'all'),
+{% if waffle.flag('no-prelim-review') %}
+    {% set NAV = ((_('Getting Started'), 'all'),
+              (_('Upload your add-on'), 'all'),
+              (_('Describe your add-on'), 'all'),
+              (_('Add images'), 'listed'),
+              (_('Select a license'), 'listed'),
+              (_('n/a'), 'hidden'),
```
Comment (diox) [inline src/olympia/devhub/templates/devhub/addons/submit/sidebar.html]: Ugh. Is there no better way?

**Comment by diox** [comment]
I don't see you adding a migration for the waffle (not 100% necessary, but good practice). Also, why a flag and not just a simple switch ?

**Comment by eviljeff** [comment]
I landed the waffle in https://github.com/mozilla/addons-server/pull/3178 - a switch seemed like a good idea at the time ... now it's easier to keep using it.

**Comment by diox** [comment]
Ah I wondered in the other PR why you had chosen that name for the waffle, had not realized it would be used for both. Carry on.

File: src/olympia/devhub/templates/devhub/addons/submit/sidebar.html (hunk 4 above)
Comment (eviljeff) [inline src/olympia/devhub/templates/devhub/addons/submit/sidebar.html]: I couldn't think of one - step 6 still needs to exist for when the waffle is off so all the code in step 7 would need to go into step 6, etc.  For a short term migration (I hope a week after we flip the waffle and update existing add-ons all the preliminary code can go) it seemed most efficient.  Suggestions welcome though.

Comment (diox) [inline src/olympia/devhub/templates/devhub/addons/submit/sidebar.html]: No you're right, it's ugly but it's probably the best solution for the time being.

**Comment by diox** [comment]
Nice work untangling this mess, I'm glad I'm not the one doing it :) r+wc

**Comment by diox** [comment]
If you feel that's too much work to rework the `check()` method tests we can delay that till the day we remove the waffle.

**Comment by eviljeff** [comment]
Reworked check a little, to handle the non-required arguments a little more clearly.  Can do more during the clean-up

**Comment by diox** [comment]
wfm

**Pull request merged**